src/cli.ts

```diff
@@ -69,12 +69,13 @@ export class CLI {
 		try {
 			await validateCli();
 		} catch (error: any) {
-			if (!(error instanceof Error)) {
+			this.valid = false;
+
+			if (!(error instanceof ValidationError)) {
 				throw error;
 			}
 
-			if (error.message.includes("executable")) {
-				this.valid = false;
+			if (error.type === "not-found") {
```

Comment:
nit: I tend to prefer `switch` statements over `if` for checks on `enum` types (I know it's a string union but that's all enums are anyway in typescript 😅), since I think it makes it more obvious and upfront that we should be checking each variant.